src/Compilers/Core/Portable/Emit/AnonymousTypeKey.cs

```diff
@@ -21,7 +21,17 @@ internal struct AnonymousTypeKeyField : IEquatable<AnonymousTypeKeyField>
         /// </summary>
         internal readonly bool IsKey;
 
-        internal AnonymousTypeKeyField(string name, bool isKey = false)
+        internal static AnonymousTypeKeyField CreateField(string name)
+        {
+            return new AnonymousTypeKeyField(name, isKey: false);
+        }
+
+        internal static AnonymousTypeKeyField CreateField(string name, bool isKey)
```

Comment:
Yes, the first overload should be used in C# and the second should be used in VB. Ideally, we'd assert that, although that might require looking at the stack.